pandas/core/base.py

```diff
@@ -473,7 +478,11 @@ def is_any_frame() -> bool:
                 # we have a dict of scalars
 
                 # GH 36212 use name only if self is a series
-                name = self.name if (self.ndim == 1) else None
+                if self.ndim == 1:
```

Comment:
You can just rewrite this to `if isinstance(self, ABCSeries):` instead of checking `ndim`. It would be more consistent with other checks in the code base and get rid of the need for a cast

Reply:
Agreed, at the expense of being a little slower. On my machine, I'm seeing 127 ns for ndim and cast together and 156 ns for isinstance.

Reply:
hmm, is that because ABCSeries resolves to `Any` and hence mypy is no longer checking this. So I guess a cast will be needed in the future when we turn on more strict checking options for Any. 

isinstance(.., ABC..) are generally slow and should normally be avoided if possible, but here it seems the ndim check is not that much more performant.

but this is OK for now.

Reply:
A 28 ns timing difference is not worth worrying about. If we wanted to instance check to be stricter we should probably just add a stub for the ABCs (or see if there's a mypy hook to strongly type them) but that is out of scope for this PR

Reply:
stubs for ABCs would be good.

but I see three issues with the current change.

1. impacted performance just to pass mypy. no matter how small
2. changing the typing to resolve to Any just to pass mypy. bad practice
3. code churn. no matter how trivial. 

all three together and i'm now firmly -1 on this approach.